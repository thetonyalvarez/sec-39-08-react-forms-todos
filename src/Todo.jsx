import { useState } from 'react'
import './Todo.css'
import Box from '@mui/material/Box'
import Button from '@mui/material/Button'
import Typography from '@mui/material/Typography'
import TableRow from '@mui/material/TableRow';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';

const Todo = ({ id, task, handleRemove }) => {

  const remove = () => handleRemove(id)

  return (
    <TableRow
      className="Todo"
      data-testid="todo-row"
    >
      <TableCell>
        <Typography variant="span">{task}</Typography>
      </TableCell>
      <TableCell padding="checkbox">
        <Button onClick={remove} data-testid="remove-btn">X</Button>
      </TableCell>
    </TableRow>
  )
}

export default Todo;