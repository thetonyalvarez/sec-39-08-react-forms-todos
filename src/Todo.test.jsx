import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import Todo from './Todo'

afterEach(cleanup)

// smoke test
it("renders without crashing", function() {
  render(<Todo />)
});

// snapshot
it("matches snapshot", function() {
  const {asFragment} = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

it("shows a new Todo", function() {
  const { getByTestId } = render(<Todo task="task test"/>)
  const todo = getByTestId('todo-row')
  
  expect(todo).toContainHTML('task test')
})