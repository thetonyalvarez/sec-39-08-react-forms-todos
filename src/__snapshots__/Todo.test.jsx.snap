// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiTypography-root MuiTypography-span css-37jdci-MuiTypography-root"
  />
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="remove-btn"
    tabindex="0"
    type="button"
  >
    X
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
